webapp/src/main/java/com/cmds/webapp/controllers/APIController.java

```diff
@@ -211,7 +206,6 @@ public int updateStatus(@PathVariable("id") String id, HttpServletRequest reques
         HashMap<String, Object> deliveryData = objectMapper.readValue(jsonData, new TypeReference<HashMap<String, Object>>() {});
         // Extract specific data from the parsed JSON
         String status = (String) deliveryData.get("status");
```

Comment:
Add all your update conditions here:
so get the status and if it is COMPLETE call `trip.getAssignedRobot().removeTrip(tripId)` or something like this. And resave the robot.